ludwig/trainers/trainer_llm.py

```diff
@@ -164,24 +197,10 @@ def train(
                 save_path,
             )
         finally:
-            # ================ Finished Training ================
-            self.callback(
-                lambda c: c.on_trainer_train_teardown(self, progress_tracker, save_path, self.is_coordinator()),
-                coordinator_only=False,
+            self.finish_training(
```

Comment:
we can find a better name for this function. Maybe saome thing more explicit of what actually happens inside it, like "close_writers" or something long those lines

Reply:
Done.